app/controllers/RemoveNirmsController.scala

```diff
@@ -65,9 +65,12 @@ class RemoveNirmsController @Inject() (
           value =>
             request.userAnswers.set(RemoveNirmsPage, value) match {
               case Success(answers) =>
-                sessionRepository.set(answers).map { _ =>
-                  Redirect(navigator.nextPage(RemoveNirmsPage, NormalMode, answers))
-                }
+                for {
```

Comment:
Ensures the value on the CYA page is correct if they click yes or no on removal of nirms confirmation page.

Reply:
Pattern here feels a bit odd

Could move the outer .set into the for yield to simplify (with better names than updatedAnswers2)

`            for {
              updatedAnswers <- Future.fromTry(request.userAnswers.set(RemoveNirmsPage, value))
              updatedAnswers2 <- Future.fromTry(
                updatedAnswers.set(HasNirmsUpdatePage, !value)
              )
              _              <- sessionRepository.set(updatedAnswers)

            } yield Redirect(navigator.nextPage(RemoveNirmsPage, NormalMode, updatedAnswers2))
`

Reply:
I'll take a look and update this comment when done :)